src/core/main.c

```diff
@@ -700,26 +700,24 @@ static int parse_config_file(void) {
                 {}
         };
 
-        _cleanup_strv_free_ char **_free_files = NULL, **_free_dirs = NULL;
-
-        const char *const *files, *const *dirs, *suffix;
+        _cleanup_strv_free_ char **files = NULL, **dirs = NULL;
+        const char *suffix;
         int r;
 
-        if (arg_system) {
-                files = STRV_MAKE_CONST(PKGSYSCONFDIR "/system.conf");
-                dirs = (const char* const*) CONF_PATHS_STRV("systemd");
+        if (arg_system)
                 suffix = "system.conf.d";
-        } else {
-                r = manager_find_user_config_paths(&_free_files, &_free_dirs);
+        else {
+                r = manager_find_user_config_paths(&files, &dirs);
                 if (r < 0)
                         return log_error_errno(r, "Failed to determine config file paths: %m");
-                files = (const char* const*) _free_files;
-                dirs = (const char* const*) _free_dirs;
+
                 suffix = "user.conf.d";
         }
 
         (void) config_parse_many(
-                        files, dirs, suffix,
+                        (const char* const*) (files ?: STRV_MAKE(PKGSYSCONFDIR "/system.conf")),
```

Comment:
iirc we don't want to use these macros that create data storage on the stack inlined in function calls either - so perhaps just unconditionally define these above in separate variables, inline with the declarations, and use them if files/dirs are empty?

Reply:
> iirc we don't want to use these macros that create data storage on the stack inlined in function calls either

I cannot remember we have such coding rule.
Maybe, you are confused with the limitation about `alloca()`?

Reply:
This patch is OK. The STRV_MAKEd object lives until the function returns, which is enough.
The original bug was that the object was declared in a scope created by the conditional. I think this is nicer code than what I wrote anyway, so let's merge it.

Reply:
mmh I have the distinct memory of a review where I'd been asked not to use STR_MAKE inline - might be misremembering though. Good to know that it's fine